app/models/user.rb

```diff
@@ -44,6 +45,17 @@ def role_exists
     errors.add(:role, 'doesnt exist')
   end
 
+  def referral_exists
+    errors.add(:referral_id, 'doesnt exist') if referral_id.present? && User.find_by(id: referral_id).blank?
```

Comment:
we use dictionary errors, so `doesnt_exist` or just `exist` - it will point FE with translations